server/src/main/java/org/elasticsearch/common/breaker/ChildMemoryCircuitBreaker.java

```diff
@@ -48,6 +57,12 @@ public ChildMemoryCircuitBreaker(BreakerSettings settings, Logger logger, Hierar
         this.logger = logger;
         logger.trace(() -> format("creating ChildCircuitBreaker with settings %s", settings));
         this.parent = parent;
+        this.trippedCountMeter = telemetryProvider.getMeterRegistry()
+            .registerLongCounter(
+                HierarchyCircuitBreakerService.metricName(name),
```

Comment:
My understanding is that, in general, the advice is to use attributes for dynamic qualities of metrics. Those attributes should also be low cardinality.  What's the cardinality of the names here? 

Please keep me honest here @stu-elastic, in case I'm giving bad advice :)

Reply:
It is a good advice and a good question to ask; in general, names can be generated but there should be no dynamic or variable content in the name: that content belongs to a dimension (attributes/labels). For example, node name, node id, cluster id, etc. are all considered dynamic content that belongs to attributes, not to the metric name.
Even when there are different "flavors" of a metric (i.e. s3, azure, etc) use an attribute rather than inserting it in the metric name. 

There may be exceptions of course: when similar metrics have significantly different implementations and different related metrics (e.g. if your s3 metrics include s3/aws specific measures and your azure metrics are specific to azure).
Also, if you have a finite and fixed set of names it might be OK to have them in the name (e.g. "young" and "old" for GC generations).

Rule of thumb: you should be able to do aggregations (e.g. sum, avg) across a dimension of a given metric (without the need to aggregate over different metric names); on the other hand, any aggregation across a any dimension of a given metric should be meaningful.

(We are building a guideline on metric name conventions)

Reply:
Circuit breakers are per node and they are limited (at least at the moment). There are three (child) circuit breakers per node and one parent circuit breaker. Anyway, whether something needs to be a dimension or part of the name of a metric depends on expectations about aggregations:
1. one metric with a name dimension means I can aggregate away the name and get a total
2. one metric per each name (so no dimension) means eventually you need to use some pattern matching/regex on the name to get totals.

That said circuit breakers are per node and aggregating them at a level that is above the node (cluster, region,...) has little to no value.

Reply:
Also if the name is used as a dimension one might think that summing or averaging counts (after taking a rate) over different circuit breakers trip count values is meaningful...while it is not.

Reply:
>  circuit breakers are per node and aggregating them at a level that is above the node (cluster, region,...) has little to no value

I'm note sure that is true. Eg getting regional statistics on circuit breaks is surely useful?

Also, remember that each one of these metric names ultimately becomes a field. So placing variable elements in the name contributes to mappings explosion (just think about the Kibana dropdown for fields when filtering on a metric and having to select from dozens or hundreds of similarly named metrics).

In this case, I think using the name is ok, as there is essentially a relatively small fixed list of circuit breakers (can you try listing all of them out to double check the total number isn't too large?). If that changes we may want to consider moving the name to an attribute. But this matches up with [the plans](https://github.com/elastic/elasticsearch/pull/100907/files#diff-47dd6465f5f9586ef1ef45eb14defce4fe103e2730f6722bb1fe8494fcf76c12R39) for threadpool metrics (the threadpool name being in the metric name).

Reply:
Considering the counter is per node does it really make sense we aggregate it if in a region or cluster we have hundreds or thousands of them? If yes then I take back what I said.

Reply:
IMO having the ability to aggregate on individual circuit breakers could allow us to detect when there is a region or deployment wide problem for a specific breaker (which likely would be caused by a problem in whatever requests use that breaker).

Reply:
I am not questioning the ability to aggregate circuit breakers trip counts in general...what I am saying is that IMO it does not make sense to do something like summing up `fielddata.trip_count` with `requests.trip_count`.

It is ok instead to have aggregation of each of them as distinct metrics. Which means if you want you can aggregate on `cloud_region` summing up `fielddata.trip_count` but not mixing it with another one, like `requests.trip_count`.

Reply:
Your reasoning makes sense. Please just make sure the name follows the naming guidelines Lorenzo has proposed. Currently the metric name looks like it uses a java package prefix. I think instead it should be something like `breaker.NAME.trip_count` or something like that (Lorenzo can advise better here).